### ISSUE TITLE
[a11y]: Tag Close Button Doesn't Announce Associated Tag

### ISSUE THREAD

**Issue by Manudeep090798** [carbon-design-system/carbon#19814]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.82.1

### Automated testing tool and ruleset

JAWS Chrome

### Description

The "Close" button on a selected filter tag does not announce the associated tag name. Currently, screen readers only announce “Close filter Button”, which is not sufficient context for users relying on assistive technology.

<img width="1460" alt="Image" src="https://github.com/user-attachments/assets/c6f73b7a-7b88-4637-9f44-23b8c941ab8a" />

### WCAG 2.1 Violation

1.3.1

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-tag--dismissible

### Steps to reproduce

1. Navigate to https://react.carbondesignsystem.com/?path=/story/components-tag--dismissible.
2. Use any screen reader of your choice to check the announcement of the close button label.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
